packages/jni-swig-stub/src/main/jni/realm_api_helpers.cpp

```diff
@@ -362,6 +362,16 @@ bool realm_should_compact_callback(void* userdata, uint64_t total_bytes, uint64_
     return result;
 }
 
+void realm_data_initialization_callback(void* userdata) {
+    auto env = get_env(true);
+    static JavaClass java_data_init_class(env, "io/realm/internal/interop/DataInitializationCallback");
+    static JavaMethod java_data_init_method(env, java_data_init_class, "invoke", "()Z");
+
+    jobject callback = static_cast<jobject>(userdata);
+    jboolean result = env->CallBooleanMethod(callback, java_data_init_method);
```

Comment:
We're not checking the `result` of the call?

Reply:
Fixed